File: server/src/main/java/org/dan/jadalnia/app/label/LabelService.java
package org.dan.jadalnia.app.label;

import lombok.RequiredArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.extern.slf4j.Slf4j;
import org.dan.jadalnia.app.festival.pojo.Festival;
import org.dan.jadalnia.app.order.pojo.OrderLabel;

import javax.inject.Inject;
import java.util.concurrent.CompletableFuture;

@Slf4j
@FieldDefaults(makeFinal = true)
@RequiredArgsConstructor(onConstructor = @__(@Inject))
public class LabelService {
    LabelDao labelDao;

    public CompletableFuture<OrderLabel> allocate(Festival festival) {
        return labelDao.storeNew(
                festival.fid(),
                festival.getNextLabel().getAndIncrement());
    }
}

File: service/src/main/kotlin/org/dan/jadalnia/app/label/LabelService.kt
package org.dan.jadalnia.app.label;

import org.dan.jadalnia.app.festival.pojo.Festival;
import org.dan.jadalnia.app.order.pojo.OrderLabel;

import javax.inject.Inject;
import java.util.concurrent.CompletableFuture;

class LabelService @Inject constructor (val labelDao: LabelDao) {
    fun allocate(festival: Festival): CompletableFuture<OrderLabel> =
        labelDao.storeNew(festival.fid(), festival.nextLabel.getAndIncrement())

}
